### PR TITLE
[grafana] Prevent admin password from changing every upgrade

### PR DESCRIPTION
**Pull request by 0xMH**
Solving #10 
Using a lookup function to check if the password secret exists or not before generating new one, If It's indeed exit It'll be reused. 

Lookup functions are [first introduced in helm3](https://helm.sh/docs/chart_template_guide/functions_and_pipelines/#using-the-lookup-function) so By introducing a new lookup function this chart is pretty much incompatible with Helm2 But at this point of time, I'm not really sure that this is a problem cause 
[On November 13, 2020, Helm v2 support will formally end](https://github.com/helm/charts#status-of-the-project)

### CODE REVIEW

**Comment by zanhsieh** [comment]
@0xMH 
Could you bump the chart version to latest please? Thank you.

**Comment by 0xMH** [comment]
@zanhsieh Can you check now please? Any other changes required?

File: charts/grafana/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: grafana
-version: 5.8.10
+version: 5.8.11
```
Comment (zanhsieh) [inline charts/grafana/Chart.yaml:3]: Actually the version shall be 5.8.13 or larger.

Comment (0xMH) [reply charts/grafana/Chart.yaml]: Done

**Comment by 0xMH** [comment]
@zanhsieh I increased the version numb again. Any more changes?

**Review by zanhsieh — APPROVED** [review]

File: charts/grafana/templates/_helpers.tpl [hunk 2]
```diff
@@ -100,3 +100,16 @@ Selector labels ImageRenderer
 app.kubernetes.io/name: {{ include "grafana.name" . }}-image-renderer
 app.kubernetes.io/instance: {{ .Release.Name }}
 {{- end -}}
+
+{{/*
+Looks if there's an existing secret and reuse its password. If not it generates
+new password and use it.
+*/}}
+{{- define "grafana.password" -}}
+{{- $secret := (lookup "v1" "Secret" (include "grafana.namespace" .) (include "grafana.fullname" .) ) -}}
```
Comment (zanhsieh) [inline charts/grafana/templates/_helpers.tpl:109]: ==> Linting charts/grafana
Error:  templates/: parse error at (grafana/templates/_helpers.tpl:109): function "lookup" not defined

**Comment by torstenwalter** [comment]
@0xMH DCO check is failing (https://github.com/grafana/helm-charts/runs/1302545388) and also linting fails

**Comment by 0xMH** [comment]
@torstenwalter The Linting error is kinda expected because the lookup func is addded on Helm3. I think we should consider to drop support form helm 2 as its EOF is already in the really near future.

File: charts/grafana/templates/_helpers.tpl (hunk 2 above)
Comment (0xMH) [reply charts/grafana/templates/_helpers.tpl:109]: Lookup func is introduced on Helm v3

**Comment by zanhsieh** [comment]
@0xMH 
Please rebase after https://github.com/grafana/helm-charts/pull/69 merged. Thank you.

@torstenwalter @maorfr 
Please help out with https://github.com/grafana/helm-charts/pull/69 .

**Comment by zanhsieh** [comment]
@0xMH 
https://github.com/grafana/helm-charts/pull/69 has been merged. please rebase from there. Thank you and sorry for waiting.

**Comment by 0xMH** [comment]
Done. Can you check now please? @zanhsieh

**Comment by torstenwalter** [comment]
@0xMH Build is failing. Can you try to specify an `image` for https://github.com/grafana/helm-charts/blob/main/.github/workflows/lint-test.yaml#L17

I think `quay.io/helmpack/chart-testing:v3.3.0` contains an updated helm version.

**Comment by 0xMH** [comment]
@torstenwalter Done.

File: .github/workflows/lint-test.yaml [hunk 3]
```diff
@@ -14,7 +14,7 @@ jobs:
 
       - name: Run chart-testing (lint)
         id: lint
-        uses: helm/chart-testing-action@v1.0.0
+        uses: quay.io/helmpack/chart-testing:v3.3.0
```
Comment (torstenwalter) [inline .github/workflows/lint-test.yaml:17]: This does not work. We have to use the chart-testing-action. Internally it's using the chart-testing image. If we want to specify a specific version then we need to do that via parameters.

**Review by torstenwalter — DISMISSED** [review]

File: .github/workflows/lint-test.yaml (hunk 3 above)
Comment (0xMH) [reply .github/workflows/lint-test.yaml:17]: @torstenwalter If we're going to migrate to helm3 shouldn't we start using [chart-testing-action@v3.0.0](https://github.com/marketplace/actions/helm-chart-testing-helm-3#example-workflow)should I update it?

**Comment by 0xMH** [comment]
I'm not sure why exactly is fails now. The linter give me `Error:  templates/: parse error at (grafana/templates/_helpers.tpl:118): unexpected <define> in command`

**Comment by sathieu** [comment]
Not stale

**Comment by sathieu** [comment]
@0xMH @zanhsieh @torstenwalter Any news on this PR?

File: charts/grafana/templates/_helpers.tpl [hunk 4]
```diff
@@ -113,4 +113,16 @@ Return the appropriate apiVersion for rbac.
 {{- else -}}
 {{- print "rbac.authorization.k8s.io/v1beta1" -}}
 {{- end -}}
```
Comment (steven-sheehy) [inline charts/grafana/templates/_helpers.tpl:115]: This is the reason for the lint errors `unexpected <define> in command`. The previous define is missing its final end block.

```suggestion
{{- end -}}
{{- end -}}
```

**Review by steven-sheehy — CHANGES_REQUESTED** [review]
Need to fix the lint and bump the chart version

**Comment by 0xMH** [comment]
I did the required changes, Hopefully we could merge now?

**Review by steven-sheehy — APPROVED** [review]
LGTM though I don't have commit rights.

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**